api4/user.go

```diff
@@ -106,10 +106,9 @@ func createUser(c *Context, w http.ResponseWriter, r *http.Request) {
 	var ruser *model.User
 	var err *model.AppError
 	if len(tokenId) > 0 {
-		var token *model.Token
-		token, err = c.App.Srv().Store.Token().GetByToken(tokenId)
-		if err != nil {
-			c.Err = model.NewAppError("CreateUserWithToken", "api.user.create_user.signup_link_invalid.app_error", nil, err.Error(), http.StatusBadRequest)
+		token, nErr := c.App.Srv().Store.Token().GetByToken(tokenId)
+		if nErr != nil {
```

Comment:
Let's check for the error here and set the correct status code.

Reply:
I understand the point but here the previous code was always returning http status `http.StatusBadRequest` even when the error was `http.StatusInternalServerError`, so these cases are not so easy to deduct, my goal is always to migrate to `error` without changing the returned http status to the caller.

For example below of this sentence http 400 still being returned when license is nil or guest account settings is not enabled.

I will do the changes but I just wanted to explain why I didn't make the changes before :)

Reply:
But here we should only change the returned http error code right? Because the message key should be kept I think @agnivade

Reply:
That is correct. Only change the status code. Thanks Rodrigo.